TAO/tests/IDLv4/maps/test.idl

```diff
@@ -1,32 +1,42 @@
 struct RecurseStruct;
 typedef map<int32, RecurseStruct> IntRecursedMap;
 struct RecurseStruct {
-    IntRecursedMap iMapR;
+  IntRecursedMap iMapR;
 };
 
 struct TestStruct {
-    int32 id;
-    string msg;
+  int32 id;
+  string msg;
 };
 
 typedef sequence<TestStruct> TestStructSeq;
 typedef map<int32, TestStruct> TestStructMap;
 typedef map<int32, string> TestIntStringMap;
 
 struct DataStruct {
-    // Bounded maps are not actually bounded as std::map doesn't support that
+  // NOTE: Bounded maps are not actually bounded as std::map doesn't support that
 
-    map<int32, int32> intIntMap;
-    // map<int32, int32, 4> intIntMapBounded;
+  map<int32, int32> intIntMap;
+  map<string, int32, 4> stringIntMapBounded;
 
-    map<string, TestStruct> stringStructsMap;
-    // map<string, TestStruct, 4> stringStructsMapBounded;
+  map<string, TestStruct> stringStructsMap;
+  map<uint32, TestStruct, 4> uint32StructsMapBounded;
 
-    map<string, TestStructSeq> stringSequenceMap;
-    // map<string, TestStructSeq, 2> stringSequenceMapBounded;
+  map<string, TestStructSeq> stringSequenceMap;
+  map<uint32, TestStructSeq, 2> uint32SequenceMapBounded;
 
-    map<string, TestStructMap> stringMapMap;
-    // map<string, TestStructMap, 2> stringMapMapBounded;
+  map<string, TestStructMap> stringMapMap;
+  map<uint32, TestStructMap, 2> uint32MapMapBounded;
 
-    map<TestIntStringMap, string> mapStringMap;
+  map<TestIntStringMap, string> mapStringMap;
```

Comment:
Does a map with a map key work?

Reply:
Yes, apparently maps have comparison operators.

Reply:
Although IDL43-70 isn't finalized yet, I don't think they'll be supported in IDL.
Not a problem with it being tested here -- but we may not end up allowing them in OpenDDS.

Reply:
This isn't a IDL issue though, it's a mapping issue and for C++ at least it would be solved by providing comparison operators.